atproto/lexicon/resolve.go

```diff
@@ -0,0 +1,62 @@
+package lexicon
+
+import (
+	"context"
+	"fmt"
+	"log/slog"
+
+	"github.com/bluesky-social/indigo/api/agnostic"
+	"github.com/bluesky-social/indigo/atproto/data"
+	"github.com/bluesky-social/indigo/atproto/identity"
+	"github.com/bluesky-social/indigo/atproto/syntax"
+	"github.com/bluesky-social/indigo/xrpc"
+)
+
+// Low-level routine for resolving an NSID to full Lexicon data record (as stored in a repository).
+//
+// The current implementation uses a naive 'getRepo' fetch to the relevant PDS instance, without validating MST proof chain.
+//
+// Calling code should usually use ResolvingCatalog, which handles basic caching and validation of the Lexicon language itself.
+func ResolveLexiconData(ctx context.Context, dir identity.Directory, nsid syntax.NSID) (map[string]any, error) {
+
+	baseDir := identity.BaseDirectory{}
+	did, err := baseDir.ResolveNSID(ctx, nsid)
+	if err != nil {
+		return nil, err
+	}
```

Comment:
Is there a reason we use the BaseDirectory here and not the identity directory? I guess to avoid adding ResolveNSID as a required method on the directory interface for now?

Reply:
Never mind I see we suggest you use `ResolvingCatalog` anyhow

Reply:
yeah, basically putting that method on `BaseDirectory` to start. would probably make sense to add to caching and other implementations, and thus the interface, but doing baby steps.